File: InterviewHub/my-react-app/src/services/progressService.js
import axiosClient from './axiosClient';

export const progressService = {
  markQuestionStatus: (data) => {
    return axiosClient.post('/progress', data);
  },
  getUserProgress: (params) => {
    return axiosClient.get('/progress', { params });
  },
  getQuestionProgress: (questionId) => {
    return axiosClient.get(`/progress/${questionId}`);
  },
  getLearnedQuestionsByCategory: (categoryId, params) => {
    return axiosClient.get(`/progress/category/${categoryId}/learned`, { params });
  }
};

File: Backend/src/main/java/com/InterviewHub/feature/progress/ProgressService.java
package com.InterviewHub.feature.progress;

import com.InterviewHub.dto.PaginationDTO;
import com.InterviewHub.feature.progress.dto.UpdateProgressRequest;
import com.InterviewHub.feature.progress.dto.UserProgressResponse;

public interface ProgressService {
    UserProgressResponse markQuestionStatus(UpdateProgressRequest request);
    PaginationDTO<UserProgressResponse> getUserProgress(int page, int size);
    UserProgressResponse getQuestionProgress(Long questionId);
    PaginationDTO<com.InterviewHub.feature.question.dto.QuestionResponse> getLearnedQuestionsByCategory(Long categoryId, int page, int size);
}
